applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -430,17 +560,18 @@ static void uart_callback(struct uart_event *evt, void *user_data)
 		LOG_INF("TX_ABORTED");
 		break;
 	case UART_RX_RDY:
-		uart_rx_handler(evt->data.rx.buf[0]);
+		for (int i = pos; i < (pos + evt->data.rx.len); i++) {
+			uart_rx_handler(evt->data.rx.buf[i]);
+		}
+		pos += evt->data.rx.len;
 		break;
 	case UART_RX_BUF_REQUEST:
-		err = uart_rx_buf_rsp(uart_dev, &uart_rx_buf[buf_num], 1);
-		if (err) {
-			LOG_WRN("UART RX buf rsp: %d", err);
-		}
-		buf_num++;
-		buf_num %= UART_RX_LEN;
+		pos = 0;
+		err = uart_rx_buf_rsp(uart_dev, next_buf,
```

Comment:
Pos is used by current buffer receiving, better a separate pointer for each buffer

Reply:
Only 1 buffer is reported back to user at a time, so one pointer should be enough.